### PR TITLE
enhance(scm/webhook): handle default branch change and archived repo

### PR DESCRIPTION
**Pull request by ecrupper**
Closes https://github.com/go-vela/community/issues/325

Closes https://github.com/go-vela/community/issues/659

Building off the work done in https://github.com/go-vela/server/pull/556, using repository events can help maintain parity between GitHub repos and their Vela equivalent.

### CODE REVIEW

**Review by cognifloyd — DISMISSED** [review]
LGTM

**Review by jbrockopp — APPROVED** [review]
LGTM

File: api/webhook.go [hunk 1]
```diff
@@ -172,14 +170,56 @@ func PostWebhook(c *gin.Context) {
...
+				retErr := fmt.Errorf("%s: failed to update repo %s: %w", baseErr, r.GetFullName(), err)
+				util.HandleError(c, http.StatusInternalServerError, retErr)
+
+				h.SetStatus(constants.StatusFailure)
+				h.SetError(retErr.Error())
+
+				return
+			}
+
+			c.JSON(http.StatusOK, fmt.Sprintf("no build to process, repository %s.", h.GetEventAction()))
```
Comment (wass3r) [inline api/webhook.go:214]: let's be more specific on what action took place here. we are making a changed in the database - we should call that out.

**Review by wass3r — CHANGES_REQUESTED** [review]
lgtm - one minor request

**Review by wass3r — APPROVED** [review]

**Review by plyr4 — APPROVED** [review]

**Pull request merged**